src/Symfony/Component/Lock/Store/LegacyStore.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Lock\BlockingStoreInterface;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\ExpiringStoreInterface;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\PersistStoreInterface;
+use Symfony\Component\Lock\StoreInterface;
+use Symfony\Component\Lock\Strategy\StrategyInterface;
+
+/**
+ * CombinedStore is a StoreInterface implementation able to manage and synchronize several StoreInterfaces.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class LegacyStore implements StoreInterface
+{
+    use LoggerAwareTrait;
+    use ExpiringStoreTrait;
+
+    /** @var PersistStoreInterface */
+    private $decorated;
+    /** @var StrategyInterface */
+    private $strategy;
+
+    /**
+     * @param PersistStoreInterface $decorated   The list of synchronized stores
+     * @param StrategyInterface $strategy
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(PersistStoreInterface $decorated)
+    {
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(Key $key)
+    {
+        $this->decorated->save($key);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function waitAndSave(Key $key)
+    {
+        if ($this->decorated instanceof BlockingStoreInterface && $this->decorated->supportsWaitAndSave()) {
+            $this->decorated->waitAndSave($key);
+        } else {
+            throw new NotSupportedException();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        if ($this->decorated instanceof ExpiringStoreInterface && $this->decorated->supportsPutOffExpiration()) {
+            $this->decorated->putOffExpiration($key, $ttl);
+        } else {
+            throw new NotSupportedException();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(Key $key)
+    {
+       $this->decorated->delete($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(Key $key)
+    {
+        $this->decorated->exists($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsPutOffExpiration(): bool
+    {
+        if ($this->decorated instanceof ExpiringStoreInterface && $this->decorated->supportsPutOffExpiration()) {
```

Comment:
`return $this->decorated instanceof ExpiringStoreInterface && $this->decorated->supportsPutOffExpiration();`